lib/ddtrace/sampling/matcher.rb

```diff
@@ -0,0 +1,59 @@
+module Datadog
+  module Sampling
+    # Checks if a span conforms to a matching criteria.
+    class Matcher
+      # Returns `true` if the span should conforms to this rule, `false` otherwise
+      #
+      # @abstract
+      # @param [Span] span
+      # @return [Boolean]
+      def match?(span)
+        raise NotImplementedError
+      end
+    end
+
+    # A \Matcher that supports matching a span by
+    # operation name and/or service name.
+    class SimpleMatcher < Matcher
+      # Returns `true` for case equality (===) with any object
+      MATCH_ALL = Class.new do
+        # DEV: A class that implements `#===` is ~20% faster than
+        # DEV: a `Proc` that always returns `true`.
+        def ===(other)
+          true
+        end
+      end.new
+
+      attr_reader :name, :service
+
+      # @param name [String,Regexp,Proc] Matcher for case equality (===) with the span name, defaults to always match
+      # @param service [String,Regexp,Proc] Matcher for case equality (===) with the service name, defaults to always match
+      def initialize(name: MATCH_ALL, service: MATCH_ALL)
+        @name = name
+        @service = service
+      end
+
+      # rubocop:disable Style/CaseEquality
```

Comment:
This is a silly Rubocop rule... fine that you disabled it here. Might want to disable it in the config later too.

Reply:
Done! Added to `.rubocop.yml`.